### PR TITLE
 [176] use SensitivityWatchEventModifier#HIGH for MacOS

### PR DESCRIPTION
**Pull request by adietish**

### CODE REVIEW

**Comment by adietish** [comment]
fixes #176

File: framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/filewatcher/FileWatcherService.java [hunk 1]
```diff
@@ -175,19 +182,39 @@ private void subscribeSinglePath(Path working) {
...
+	private WatchKey register(Path working) throws IOException {
+		WatchEvent.Kind<?>[] watchEventKinds = new WatchEvent.Kind<?>[] { 
+			StandardWatchEventKinds.ENTRY_CREATE,
+			StandardWatchEventKinds.ENTRY_DELETE, 
+			StandardWatchEventKinds.ENTRY_MODIFY };
+		HighSensitivityWatchEventModifier highSensivity = new HighSensitivityWatchEventModifier();
+
+		if (highSensivity.isRequired()) {
+			if (!highSensivity.exists()) {
+				LOG.warn("Watchservice requires high sensitivity on this system."
```
Comment (adietish) [inline framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/filewatcher/FileWatcherService.java]: In this patch we're logging a warning if we see high sensitivity being necessary but not available. Maybe we should provide ways for the UI to present a dialog to the user in this case.
@robstryker @jeffmaury thoughts?

Comment (jeffmaury) [reply framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/filewatcher/FileWatcherService.java]: The only issue is that the message will be logged only when the service will be used

**Comment by adietish** [comment]
Although the build is green, things are still not ok on MacOS:
FilewatcherModificationTest is running fine standalone in Eclipse but is not in maven:
```
Results :

Failed tests:
  FilewatcherModificationsTest.testFileWatcherEvent_subscribeDeleteRecreate:193
  FilewatcherModificationsTest.testFileWatcherEvent_subscribeRootDeleteRootRecreateDeep:239

Tests run: 76, Failures: 2, Errors: 0, Skipped: 0
```
@jeffmaury @robstryker ideas?

**Comment by adietish** [comment]
I managed to have tests running fine in Maven as soon as I add a wait for the latches one I deleted and  recreated the root folder.

**Comment by adietish** [comment]
@jeffmaury @robstryker I now added additional waits for the latches if the tests are run on mac and also increase the timeout for the watches for macs only.
I still believe that we should not offer functionality in ```FileWatcherService``` that's not reliable for Mac. We should think hard about restricting it to the reliable functionality ```something changed``` vs the current ```file/folder X changed```.

**Comment by robstryker** [comment]
testPR

**Pull request merged**